retrieval/discovery/consul.go

```diff
@@ -95,6 +96,15 @@ func NewConsulDiscovery(conf *config.ConsulSDConfig) *ConsulDiscovery {
 		scrapedServices: map[string]struct{}{},
 		services:        map[string]*consulService{},
 	}
+	if clientConf.Datacenter == "" {
+		info, err := client.Agent().Self()
+		if err != nil {
+			panic(fmt.Errorf("discovery.NewConsulDiscovery: %s", err))
+		}
+		cd.clientDatacenter = info["Config"]["Datacenter"].(string)
+	} else {
```

Comment:
@brian-brazil just saw your note, yeah I did a `git push --force`. I can [see it here](http://i.rxt.gd/15H6K) in the files changed view but not in the conversation view for some reason. I can also see it in my [fork](https://github.com/robbiet480/prometheus/blob/80b41f7bcd46bd39987b6646cf93c2d9e08c9928/retrieval/discovery/consul.go#L99)